src/Map.php

```diff
@@ -3727,4 +3727,32 @@ protected function visit( iterable $entries, array &$result, int $level, ?\Closu
 			}
 		}
 	}
+
+
+	/**
+	 * Executes $callback if the Map is empty
+	 *
+	 * Examples: (Returns match or dies if doesn't exist)
+	 *  $page = $this->pages
+	 *          ->filter(fn(Page $page, $_) => $page->Id == $pageId)
+	 *          ->if(
+	 *              fn(Map $Map) => $Map->isEmpty(),
+	 *              fn(Map $_) => die("unknown page: " . $pageId))
+	 *          ->first();
+	 *
+	 * @param callable|bool $condition Bool or function with (Map) parameter which evaluates to bool
+	 * @param callable $then Function with (Map) parameter which returns void
+	 * @param callable|null $else Optional function with (Map) parameter which returns void
+	 * @return self Same map for fluid interface
+	 */
+	public function if( $condition, callable $then, callable $else = null ): self
```

Comment:
The second and third arguments can't be a callable like "strcmp", they must be a \Closure instead

Reply:
resolved in e09020e4f6c9f0232299402636f73131868c0f93